Issutracker : #i33228#, #i34210#
CWS         : N/A
Author      :
Description : Kannada doesn't build since some resources are
limited to 64k. it won't fix for 1.1.x officially

Index: inc/sc.hrc
===================================================================
RCS file: /cvs/sc/sc/inc/sc.hrc,v
retrieving revision 1.35
retrieving revision 1.35.288.2
diff -u -r1.35 -r1.35.288.2
--- sc/inc/sc.hrc	3 Dec 2002 15:26:06 -0000	1.35
+++ sc/inc/sc.hrc	17 Sep 2004 14:25:38 -0000	1.35.288.2
@@ -1141,8 +1141,12 @@
 #define RID_TBXCTL_INSOBJ		(SC_RESOURCE_START+40)
 #define RID_BMP_REFBTN1			(SC_RESOURCE_START+41)
 #define RID_BMP_REFBTN2			(SC_RESOURCE_START+42)
-#define RID_SC_FUNCTION_DESCRIPTIONS1	(SC_RESOURCE_START+43)
-#define RID_SC_FUNCTION_DESCRIPTIONS2	(SC_RESOURCE_START+44)
+// #i34210# split function resources due to 64k limit, don't reuse old values
+// to have obviously empty function lists in case some localisation used the
+// old values for any reason. Don't even reuse the exact name because of the
+// translation database IDs.
+//#define RID_SC_FUNCTION_DESCRIPTIONS1	(SC_RESOURCE_START+43)
+//#define RID_SC_FUNCTION_DESCRIPTIONS2	(SC_RESOURCE_START+44)
 #define RID_POPUP_CONTROL		(SC_RESOURCE_START+45)
 #define RID_POPUP_PAGEBREAK		(SC_RESOURCE_START+46)
 #define RID_OBJECTBAR_DRAWFORM	(SC_RESOURCE_START+47)
@@ -1158,6 +1162,10 @@
 #define RID_IMAGELIST_H_NAVCONT	(SC_RESOURCE_START+57)
 #define RID_OUTLINEBITMAPS_H    (SC_RESOURCE_START+58)
 #define RID_OBJECTMENU_EDIT     (SC_RESOURCE_START+59)
+#define RID_SC_FUNCTION_DESCRIPTIONS_1  (SC_RESOURCE_START+60)
+#define RID_SC_FUNCTION_DESCRIPTIONS_2  (SC_RESOURCE_START+61)
+#define RID_SC_FUNCTION_DESCRIPTIONS_3  (SC_RESOURCE_START+62)
+#define RID_SC_FUNCTION_DESCRIPTIONS_4  (SC_RESOURCE_START+63)
 
 #define STR_START               (SC_RESOURCE_START+100)
 #define STR_ROWHEIGHT			(STR_START)
Index: source/core/data/global.cxx
===================================================================
RCS file: /cvs/sc/sc/source/core/data/global.cxx,v
retrieving revision 1.31
retrieving revision 1.31.166.2
diff -u -r1.31 -r1.31.166.2
--- sc/source/core/data/global.cxx	17 Apr 2003 09:54:45 -0000	1.31
+++ sc/source/core/data/global.cxx	17 Sep 2004 14:25:38 -0000	1.31.166.2
@@ -1475,10 +1475,12 @@
 	USHORT i,j;
 	USHORT nDescBlock[] =
 	{
-		RID_SC_FUNCTION_DESCRIPTIONS1,
-		RID_SC_FUNCTION_DESCRIPTIONS2
+		RID_SC_FUNCTION_DESCRIPTIONS_1,
+		RID_SC_FUNCTION_DESCRIPTIONS_2,
+		RID_SC_FUNCTION_DESCRIPTIONS_3,
+		RID_SC_FUNCTION_DESCRIPTIONS_4
 	};
-	const USHORT nBlocks = sizeof(nDescBlock) / sizeof(USHORT);
+	const USHORT nBlocks = sizeof(nDescBlock) / sizeof(nDescBlock[0]);
 
 	aFunctionList.Clear();
 
Index: source/ui/src/scfuncs.src
===================================================================
RCS file: /cvs/sc/sc/source/ui/src/scfuncs.src,v
retrieving revision 1.76.22.8
retrieving revision 1.76.22.8.4.2
diff -u -r1.76.22.8 -r1.76.22.8.4.2
--- sc/source/ui/src/scfuncs.src	10 Sep 2004 14:45:52 -0000	1.76.22.8
+++ sc/source/ui/src/scfuncs.src	17 Sep 2004 14:25:39 -0000	1.76.22.8.4.2
@@ -93,10 +93,16 @@
  // RID_SC_FUNCTION_DESCRIPTIONS1
  // RID_SC_FUNCTION_DESCRIPTIONS2
 
+ // #i34210# again splitted resources due to 64k limit
+ // RID_SC_FUNCTION_DESCRIPTIONS_1
+ // RID_SC_FUNCTION_DESCRIPTIONS_2
+ // RID_SC_FUNCTION_DESCRIPTIONS_3
+ // RID_SC_FUNCTION_DESCRIPTIONS_4
+
 #include "scfuncs.hrc" // ID_FUNCTION_GRP_XXX, HID_FUNC_XXX
 #include "compiler.hrc" // SC_OPCODE_XXX
 
-Resource RID_SC_FUNCTION_DESCRIPTIONS1
+Resource RID_SC_FUNCTION_DESCRIPTIONS_1
 {
 	 // -=*# Resource for function DBANZAHL #*=-
 	Resource SC_OPCODE_DB_COUNT
@@ -15963,6 +15969,11 @@
 			Text[ galician ] = "Valor 1; Valor 2;... son de 1 a 30 argumentos que representan os pagamentos e ingresos.";
 		};
 	};
+};
+
+
+Resource RID_SC_FUNCTION_DESCRIPTIONS_2
+{
 	 // -=*# Resource for function IKV #*=-
 	Resource SC_OPCODE_IKV
 	{
@@ -32174,7 +32185,8 @@
 	};
 };
 
-Resource RID_SC_FUNCTION_DESCRIPTIONS2
+
+Resource RID_SC_FUNCTION_DESCRIPTIONS_3
 {
 	 // -=*# Resource for function DBANZAHL2 #*=-
 	 // -=*# Resource for function HÄUFIGKEIT #*=-
@@ -49230,6 +49242,11 @@
 			Text[ galician ] = "é o n—ãro de graos de liberdade.";
 		};
 	};
+};
+
+
+Resource RID_SC_FUNCTION_DESCRIPTIONS_4
+{
 	 // -=*# Resource for function STANDARDISIERUNG #*=-
 	Resource SC_OPCODE_STANDARD
 	{
